src/wlapplication.cc

```diff
@@ -1271,25 +1271,104 @@ void WLApplication::handle_commandline_parameters() {
 		localedir_ = commandline_["localedir"];
 		commandline_.erase("localedir");
 	}
+
+	auto checkdatadirversion = [](const std::string& dd) {
+		try {
+			std::unique_ptr<FileSystem> fs(&FileSystem::create(dd));
+			if (!fs.get()) {
+				return std::string("Unable to allocate filesystem");
+			}
+
+			size_t len;
+			void* textptr = fs->load("datadirversion", len);
+			std::string text(static_cast<char*>(textptr), len);
+			free(textptr);
+
+			size_t sep_pos = text.find('\n');
```

Comment:
This does not work on Windows, because the line end is \r\n there...
```
[00:00:00.002 real] ERROR: Unable to detect the datadir. Please specify a datadir explicitly
[00:00:00.002 real] ERROR: with the --datadir command line option. Tried the following 1 path(s):
[00:00:00.002 real] ERROR:  · 'C:\Users\*****\AppData\Local\Widelands\./data': Incorrect version string part
```